### DISCUSSION TITLE
Version 2.0, development notes.

### DISCUSSION THREAD

**Discussion by Skamer** [Skamer/Syling-Tracker#78] [Announcements]
Hello,

I bring some news about the addon. 

As you may have noticed, the addon has not been updated for quite some time. The reason is i am focused on developing the version 2, and honestly i don't have the energy to implement features on the live version for at final, redevelop again as the two version are radically different on the backend. 

The new version has been developed entirely from scratch. The core has been completely rewritten, and since it's the foundation, all other modules had to be updated. 

### Why rewrite the core ?

Simply providing customization was tedious, not maintainable and not extensible. This limitation posed a problem and motivated me to make this decision.

The new core has been developed, taking into account this experience. New tools have been created. For example, now i can easily expose any parameter of an element through the options. The saving and updating of elements concerned  are now entirely automated and managed by this new core. 

### What are others changes ?

Some adjustments have been done in the addon visual

![image](https://github.com/Skamer/Syling-Tracker/assets/849450/87b08a3a-e743-4f50-8e66-e52fe8525d9e)

Here is a non-exhaustive list.
- The trackers can be minimized from a button.
- The tracker can be automatically hidden if no content is inside.
- The contents can be minimized.
- New contents implemented Collections, Recipe Tracker and Activities.
- The POI icons have been implemented.
- New Item Bar with new options: row, column, item size, orientation, padding and margin. 
- A new Mythic + module: New visual and support of the addon Mythic Dungeon Tools for displaying the pull percent.
- And others..

### When it's ready ?

While i would like to transition you to this new version as quickly as possible, a release version is far from ready. There is still a lot of work to be done.

Nevertheless, i plan to publish an alpha or beta version on Curseforge  this month for volunteers. If you really want to test now, you can go in the "dev" branch of the repository. Each commit in this branch generates a package for the addon. 

In case where you want to post a ticket about it, don't forget to specify me the version you use.

**Comment by rknauber** [comment]
Having some tracker customization or just a general replacement is something I've wanted for years but never found anything that worked well for me. Would be happy to try and help test things when you think the alpha/beta versions are more ready for testing.

**Comment by RAtwood90** [comment]
Good to Hear Bro! Looking forward to have finally Customization Options to Edit Font, Font Scale, Font Color, Outline and more! 

Stay well!

**Comment by Fendoran** [comment]
Hi, I'm having problems understanding what to download to test it, the one I did download is not loading up inn game?

**Reply by Skamer** [reply]
1. Go in the "Actions" tab
![image](https://github.com/Skamer/Syling-Tracker/assets/849450/711e7ce9-75d6-4cff-bcdf-95be14e85cd3)

2. Then click on the more recent build in the "dev" branch
![image](https://github.com/Skamer/Syling-Tracker/assets/849450/e06eaca1-7d30-44f3-b1a7-8d5c1d2e3c31)

3. After, you can download the final package under the "Artifacts" section. NOTE: You need to be login to Github for download it.
![image](https://github.com/Skamer/Syling-Tracker/assets/849450/a7eb2d89-98ef-430f-8b24-0a82fcf9c6ef)

**Comment by Fendoran** [comment]
That worked, awesome, after testing it abit, I can very well see the potential, it's already way better than the old one, appart from the unfinished part and the wast majority of buggs, but I love what you're doing and the way you're going!

**Comment by Skamer** [comment]
There is still a lot of things needing to be iterated, especially for the options part. The difficulty with these is to find a certain hierarchy and visibility so that you don't get lost in it, and find what interests you very quickly.

**Reply by Fendoran** [reply]
I think being able to customize everything is a huge plus, all fonts types/sizes/colors, background colors of everything separately, how far down the quests goes, how wide you want it, etc

and a huge plus would be to have the quest items/spells being integrated into the quests it belongs to, instead of having it a standalone would come a long way too.

**Reply by Skamer** [reply]
For quest items, i would like to integrate them directly into the quests but unfortunately i cannot do this without causing severe problems to the tracker. This dues to Blizzard restrictions, not something i can resolve from my side.

**Reply by Fendoran** [reply]
I know GW2 UI did this though, could check it to see if that's the case :) would make it very clean looking ^^

**Reply by Skamer** [reply]
I take a look on GW2 UI, he seems to take a similar approach which is not integrating the item action button in the scrolling content, and moving them in a separate action bar.

I continue my previous answer with more details.

The reason is i don't want integrating ["protected" frame](https://wowpedia.fandom.com/wiki/Secure_Execution_and_Tainting#Protected_frames_and_secure_templates) in the tracker. 

If i want to create an "use quest item" button, i'm forced to use them. The problem with protected frame are they come with restrictions: you cannot affecting their position, visibility, interactivity or attribute while you are in combat. There are still ways to do that but under very limited condition. 

In addition of that, there is an another annoying behavior, all the frames anchored to them become protected frames, and this apply recursively. In our case, this will say the tracker and all its  content will become protected. This would add more complexity in the code, and the scrolling feature wouldn't work while you are in combat.  

I think you would get more cons than pos.

**Reply by Fendoran** [reply]
I see, well, you can make the icon appear there, but not clickable then right? If so that's good enough I guess

**Reply by Skamer** [reply]
Strange, this is the current behavior, an icon should appear to the right inside the quest. If it's not the case, it's a bug.

**Reply by Fendoran** [reply]
Yeah, that's atleast how it was in previous version, haven't had a quest with the reworked one to check it :P

**Reply by Skamer** [reply]
In the new version, this is displayed as below if the quest has an item. The cooldown is also displayed. 

![image](https://github.com/Skamer/Syling-Tracker/assets/849450/f5cf8b3a-c0e5-4fe7-9cf2-8c59d1bc2069)

**Reply by Fendoran** [reply]
I honestly don't like the basic blizzard icon you got in the new version, as displayed on that img the (2) icon.

**Reply by Skamer** [reply]
The (2) icon is a POI Button. Do you want to remove it or you want keep it but simply the design you don't like ?

**Reply by Fendoran** [reply]
I think it should be a preferance, maybe add a setting to enable/disable it, I personally don't like it

**Reply by Fendoran** [reply]
Maybe you could even edit the look of it to more match the theme and layout of your addon aswell, so ot could better match the aestetics :)

**Reply by Skamer** [reply]
I can add an setting for disabling the poi buttons. 
For style, i have retaken the blizzard art for gaining time. I'll review that later but probably need to create new textures.

**Reply by Fendoran** [reply]
Thanks man, appreciate your willingness to hear feedback :) there's a lot of stuff I can suggest to you that is very minor additions, but very welcoming and aestetically pleasing if you would be willing to discuss it in more detail :)

I'm super pumped about your addon, it's been by far the nicest looking one out there, the only thing was settings that it was missing, but with this re-worked one, I can honestly say I'm thrilled!

**Comment by Spydawg** [comment]
Option to show quests per zone?

**Reply by Skamer** [reply]
Still thinking how to implement that:
- Should filter the quests displayed in the "Campaign Quests" or "Quests" group. 
- or create a new group "Zone Quests" with the nearby quests.

**Comment by sidain** [comment]
So is the macro /slt toggle tracker main
no longer viable?  Is there any equivalent to toggle on and off?

**Reply by Skamer** [reply]
This macro is missing on the new version, i'm going to re-add it.

**Reply by Skamer** [reply]
now added as /slt tshow tracker main

**Comment by sidain** [comment]
Appreciate it. Thank you.
